### PR TITLE
documentation: S3C-4371-api-compatibility-matrix

### PR DESCRIPTION
**Pull request by anurag4DSB**
API compatibility matrix for OSIS.

This will help POs plan the next milestone and also help devs prioritize development.

### CODE REVIEW

**Review by ilkescality — APPROVED** [review]

File: docs/api-compatibility-matrix.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+| updateCredentialStatus | . | . | . |
+| deleteCredential | . | . | . |
+| getUsage | o | o | o |
+| getBucketList | . | . | . |
+| getBucketLoggingId | . | . | . |
+| getAnonymousUser | . | . | . |
+| getConsole | . | . | . |
+| getS3Capabilities * | o | x | x |
+| getInfo * | o | o | o |
+| refreshToken *  | . | . | . |
```
Comment (saitharunthodupunuru) [inline docs/api-compatibility-matrix.md:52]: This one came implemented with default stub and there needs no additional changes for this

File: docs/api-compatibility-matrix.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+| listCredentials * | . | x | x |
+| createCredential * | . | x | x |
+| getCredential * | . | . | . |
+| updateCredentialStatus | . | . | . |
+| deleteCredential | . | . | . |
+| getUsage | o | o | o |
+| getBucketList | . | . | . |
+| getBucketLoggingId | . | . | . |
+| getAnonymousUser | . | . | . |
+| getConsole | . | . | . |
```
Comment (saitharunthodupunuru) [inline docs/api-compatibility-matrix.md:49]: `getConsole` is partially implemented to always provide only Provider Console URL irrespective of input (https://scality.atlassian.net/browse/S3C-3545) . Returning "Tenant Console URL" i.e., in our case S3 Console URL, based on the input is yet to be implemented (https://scality.atlassian.net/browse/S3C-3546)

File: docs/api-compatibility-matrix.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+| createTenant * | x | x | x |
+| queryTenants * | x | x | x |
+| queryUsers * | . | x | x |
+| createUser * | . | x | x |
+| listUsers * | . | x | x |
+| getUserWithId * | . | x | x |
+| headUser | . | . | . |
+| updateUserStatus * | . | . | . |
+| deleteUser * | . | . | . |
+| getUserWithCanonicalID * | . | x | x |
```
Comment (saitharunthodupunuru) [inline docs/api-compatibility-matrix.md:38]: `getUserWithCanonicalID` is not implemented in Milestone 2

**Comment by saitharunthodupunuru** [comment]
Please add `api-compatibility-matrix.md` as a link on `docs/milestone-1.md`.
For future reference, this file needs to be added as a link `docs/milestone-2.md` for https://scality.atlassian.net/browse/S3C-4319

**Review by saitharunthodupunuru — CHANGES_REQUESTED** [review]
Left review comments for some minor changes

File: docs/api-compatibility-matrix.md (hunk 2 above)
Comment (anurag4DSB) [reply docs/api-compatibility-matrix.md]: Changed in [`0b605da` (#47)](https://github.com/scality/vmware-ose-scality/pull/47/commits/0b605da507f27f5507374d44db888ddbf3effdf5)

File: docs/api-compatibility-matrix.md (hunk 3 above)
Comment (anurag4DSB) [reply docs/api-compatibility-matrix.md]: changed in [`0b605da` (#47)](https://github.com/scality/vmware-ose-scality/pull/47/commits/0b605da507f27f5507374d44db888ddbf3effdf5)

**Comment by anurag4DSB** [comment]
> Please add `api-compatibility-matrix.md` as a link on `docs/milestone-1.md`.
> For future reference, this file needs to be added as a link `docs/milestone-2.md` for [scality.atlassian.net/browse/S3C-4319](https://scality.atlassian.net/browse/S3C-4319)

Added.
On another note, this repository needs a markdown linter. In the Milestone1.md doc these are the errors:
Screenshot 2021-05-19 at 12 22 54

**Comment by saitharunthodupunuru** [comment]
I don't understand why we need this api matrix on wiki and copied to milestone document after UAT.
As per the planning meeting, my understanding is first we will put "Validation steps" on wiki to accommodate frequent changes during UAT and will be added to milestone document after UAT is completed. 
This api matrix is about the APIs that are already planned and implemented and there should not be any change per milestone (definitely nothing changing in Milestone 1).
I suggest this api matrix to be added to markdown files directly instead of wiki.

**Comment by anurag4DSB** [comment]
> I don't understand why we need this api matrix on wiki and copied to milestone document after UAT.
> As per the planning meeting, my understanding is first we will put "Validation steps" on wiki to accommodate frequent changes during UAT and will be added to milestone document after UAT is completed. 
> This api matrix is about the APIs that are already planned and implemented and there should not be any change per milestone (definitely nothing changing in Milestone 1).
> I suggest this api matrix to be added to markdown files directly instead of wiki.

@saitharunthodupunuru  you were referring to https://scality.atlassian.net/browse/S3C-4319

I was talking about validation steps as you had mentioned https://scality.atlassian.net/browse/S3C-4319. 

The api matrix is in markdown as per this PR. Not sure what you mean.

**Comment by saitharunthodupunuru** [comment]
> > I don't understand why we need this api matrix on wiki and copied to milestone document after UAT.
> > As per the planning meeting, my understanding is first we will put "Validation steps" on wiki to accommodate frequent changes during UAT and will be added to milestone document after UAT is completed. 
> > This api matrix is about the APIs that are already planned and implemented and there should not be any change per milestone (definitely nothing changing in Milestone 1).
> > I suggest this api matrix to be added to markdown files directly instead of wiki.
> 
> @saitharunthodupunuru  you were referring to https://scality.atlassian.net/browse/S3C-4319
> 
> I was talking about validation steps as you had mentioned https://scality.atlassian.net/browse/S3C-4319. 
> 
> The api matrix is in markdown as per this PR. Not sure what you mean. 

Hey Sorry.
Confused that we were not adding api matrix to milestone document too.

**Review by saitharunthodupunuru — APPROVED** [review]

**Pull request merged**